FightPandemics/Features/Post/CreatePost/OfferRequest/OfferRequest.swift

```diff
@@ -0,0 +1,135 @@
+        closeBtn.frame.size.height = 45
+        closeBtn.clipsToBounds = true
+        closeBtn.backgroundColor = UIColor.clear
+        closeBtn.layer.masksToBounds = true
+        closeBtn.layer.cornerRadius = closeBtn.frame.size.height / 2
+        closeBtn.layer.borderColor = UIColor(hexString: "#425AF2").cgColor
+        closeBtn.layer.borderWidth = 1
+        self.addSubview(closeBtn)
+        confirmBtn.setAttributedTitle(NSAttributedString(string: "OfferRequestConfirmBtn".localized, attributes: [NSAttributedString.Key.font: Fonts.poppinsRegular.customFont(size: 15), NSAttributedString.Key.foregroundColor: UIColor(hexString: "#425AF2")]), for: .normal)
+        confirmBtn.frame.size.width = ((UIScreen.main.bounds.width - 63) / 2)
+        confirmBtn.frame.size.height = 45
+        confirmBtn.clipsToBounds = true
+        confirmBtn.backgroundColor = UIColor.clear
+        confirmBtn.layer.masksToBounds = true
+        confirmBtn.layer.cornerRadius = confirmBtn.frame.size.height / 2
+        confirmBtn.layer.borderColor = UIColor(hexString: "#425AF2").cgColor
+        confirmBtn.layer.borderWidth = 1
+        self.addSubview(confirmBtn)
+    }
+    func setUpConstraints () {
+        self.translatesAutoresizingMaskIntoConstraints = false
+        topLine.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint(item: topLine, attribute: .centerX, relatedBy: .equal, toItem: self, attribute: .centerX, multiplier: 1, constant: 23).isActive = true
+        NSLayoutConstraint(item: topLine, attribute: .top, relatedBy: .equal, toItem: self, attribute: .top, multiplier: 1, constant: 12).isActive = true
+        NSLayoutConstraint(item: topLine, attribute: .width, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1, constant: 116).isActive = true
+        NSLayoutConstraint(item: topLine, attribute: .height, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1, constant: 4).isActive = true
+        questionLbl.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint(item: questionLbl, attribute: .centerX, relatedBy: .equal, toItem: self, attribute: .centerX, multiplier: 1, constant: 23).isActive = true
+        NSLayoutConstraint(item: questionLbl, attribute: .top, relatedBy: .equal, toItem: self, attribute: .top, multiplier: 1, constant: 37).isActive = true
+        NSLayoutConstraint(item: questionLbl, attribute: .width, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1, constant: 294).isActive = true
+        NSLayoutConstraint(item: questionLbl, attribute: .height, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1, constant: 22).isActive = true
+        offerBtn.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint(item: offerBtn, attribute: .centerX, relatedBy: .equal, toItem: self, attribute: .centerX, multiplier: 1, constant: 0).isActive = true
```

Comment:
The constraints for the offer buttons are broken. I don't think we need **centerX** constraints since we already set the **leading** and **trailing** constraints. Also, we don't need the width constraint as it creates ambiguity. For the right, you have to use the negative of the number **-23**. The same thing applies to the **request button**.
I tried testing it out but the constraints are broken.

Reply:
Thanks, I'll try to fix this one too.

Reply:
really helpful